Update automation page with unlisted behaviors (#333)
OpenCTI-Platform/docs: docs/usage/automation.md

#### Enrich through connector

Will send the received STIX 2.1 bundle to the stated enrichment connector and send out the modified bundle.

> ##### Specifities of the component

Reviewer: ```suggestion
##### Specificities of the component

```

**Step will fail if a single entity does not match the connector scope**
Entities that are passed to this stage must be compatible with the enrichment connector being used, as otherwise the playbook may stop at this stage. 
The best approach is to use a first playbook that will **flag your data, by applying a specific label through manipulate knowledge step** and have another playbook listening to entities/observables being created or edited with this specific label to apply the corect enrichment connector through a decision tree. The *Reduce Knowledge* component can be used to filter out incompatible entities from being passed to this stage.

> **Step will fail if an observable triggered by the enrichment is not foud**

Reviewer: ```suggestion
**Step will fail if an observable triggered by the enrichment is not found**
```

> If the playbook stops form time to time at this step it might be because the observable (or entity) is not found in the third party system in charge of enriching the observable (or entity).

Reviewer: ```suggestion
If the playbook stops from time to time at this step, it might be because the observable (or entity) is not found in the third-party system in charge of enriching the observable (or entity).
```

#### Manipulate knowledge

> Will add, replace or remove compatible attribute of the entities contains in the received STIX 2.1 bundle and send out the modified bundle.

Reviewer: ```suggestion
Will add, replace, or remove compatible attributes of the entities contained in the received STIX 2.1 bundle and send out the modified bundle.
```

Author: Apply changes suggested by Alice.

By default, modification is applied to entities having triggered the playbook. You can toggle the corresponding option to apply it to all elements in the bundle (elements that might result from enrichment for example).

> ##### Specifities of the component

Reviewer: ```suggestion
##### Specificities of the component

```

**Remove & replace operations:**

> These operation will only remove and & replace data that have been added in the context of this playbook. This step cannot be used to modify attributes of an entity (or observable) that has already been written in your platform. For instance, if an entity is ingested with a TLP:RED marking and a specific label and your playbook listens to any entity created entity with this specific label, if you have a step remove TLP:RED marking within your playbook, this step will not be executed because the TLP:RED would be already written in your platform.

Reviewer: ```suggestion
These operations will only remove and & replace data that has been added in the context of this playbook. This step cannot be used to modify attributes of an entity (or observable) that has already been written in your platform. For instance, if an entity is ingested with a TLP:RED marking and a specific label, and your playbook listens to any entity created with this specific label, if you have a step to remove TLP:RED marking within your playbook, this step will not be executed because the TLP:RED marking would already be written in your platform.
```

**Routes:**

> - Unmodified: because of the above reason, if none of your entities contained in your stix bundle have been modified, then the stix bundle will follow will follow the **unmodified** route.

Reviewer: ```suggestion
- Unmodified: because of the above reason, if none of your entities contained in your STIX bundle have been modified, then the STIX bundle will follow the **unmodified** route.
```

> - Out: if at least one of the entity or observable of your stix bundle has been successfully modified by the manipulate knowledge step, then the stix bundle will follow the **Out** route.

Reviewer: ```suggestion
- Out: if at least one of the entity or observable of your STIX bundle has been successfully modified by the manipulate knowledge step, then the STIX bundle will follow the **Out** route.
```

#### Container wrapper

Will modify the received STIX 2.1 bundle to include the entities into an container of the type you configured. 
By default, wrapping is applied to entities having triggered the playbook. You can toggle the corresponding option to apply it to all elements in the bundle (elements that might result from enrichment for example).

> ##### Specifities of the component

Reviewer: ```suggestion
##### Specificities of the component

```

**Case Templates**
It is possible to add a case template to the container wrapper step. If you select a Case in the "container type", then the case template field will be enabled, allowing you to create a case with some tasks associated.

**Create a new container at each run**

> You can decide whether you want to create a container each time this step is trigger: by default, this is disabled, so that you can consolidate information within the same container if some update occurs.

Reviewer: ```suggestion
You can decide whether you want to create a container each time this step is triggered: by default, this is disabled, so that you can consolidate information within the same container if some update occurs.
```

**Specific situation: wrap an incident into a case**

> When the primary entity you listen to is an incident & then use the "Container Wrapper" step to create a case out of your incident, by default you case will reuse some of the attributes of your incident: 

Reviewer: ```suggestion
When the primary entity you listen to is an incident & then use the "Container Wrapper" step to create a case out of your incident, by default, your case will reuse some of the attributes of your incident: 
```

- Resolve observables an indicator is based on (add in bundle): retrieve all observables linked to the bundle's indicator from the database,
- Resolve container references (add in bundle): will add to the bundle all the relationships and entities the container contains (if the entity having triggered the playbook is not a container, the output of this component will be empty),
- Resolve neighbors relations and entities (add in bundle): will add to the bundle all relations of the entity having triggered the playbook, as well as all entities at the end of these relations, i.e. the "first neighbors" (if the entity is a container, the output of this component will be empty).
- Resolve containers containing the entity (add in bundle): will add to the bundle all the containers containing the entity or observable that triggered the playbook.

> ##### Specifities of the component

Reviewer: ```suggestion
##### Specificities of the component

```

**It is not possible to chain two rules to apply a second rule on the result of the first one:**

> For instance the following operation will not work within a playbook: listen a report creation, add everything within a bundle, then apply another rule to Resolve neighbors relations and entities (add in bundle) of all entities & observables of the container. As a result, your playbook will stop there.

Reviewer: ```suggestion
For instance, the following operation will not work within a playbook: listen to a report creation, add everything within a bundle, then apply another rule to resolve neighbor relations and entities (add in bundle) of all entities & observables of the container. As a result, your playbook will stop there.
```

**Routes:**

> - Unmodified: because of the above reason, if none of your entities or observables contained in your stix bundle have been impacted by the rule, then the stix bundle will follow will follow the **unmodified** route.

Reviewer: ```suggestion
- Unmodified: because of the above reason, if none of your entities or observables contained in your STIX bundle have been impacted by the rule, then the STIX bundle will follow will follow the **unmodified** route.
```

> - Out: if at least one of the entity or observable of your stix bundle has been successfully impacted by the rule, then the stix bundle will follow the **Out** route.

Reviewer: ```suggestion
- Out: if at least one of the entity or observable of your STIX bundle has been successfully impacted by the rule, then the STIX bundle will follow the **Out** route.
```

You can also add all indicators and relationships generated by this component in the entity having triggered the playbook, if this entity is a container.

##### Specifities of the component

**Routes:**

> - Unmodified: if none of your observables has triggered an indicator creation, then the stix bundle will follow will follow the **unmodified** route.

Reviewer: ```suggestion
- Unmodified: if none of your observables have triggered an indicator creation, then the STIX bundle will follow the **unmodified** route.
```

> - Out: if at least one of your observable of your stix bundle has triggered an indicator creation, then the stix bundle will follow the **Out** route.

Reviewer: ```suggestion
- Out: if at least one of your observables of your STIX bundle has triggered an indicator creation, then the STIX bundle will follow the **Out** route.
```

You can also add all observables and relationships generated by this component in the entity having triggered the playbook, if this entity is a container.

##### Specifities of the component

**Routes:**

> - Unmodified: if none of your indicator has triggered an observable creation, then the stix bundle will follow will follow the **unmodified** route.

Reviewer: ```suggestion
- Unmodified: if none of your indicators have triggered an observable creation, then the STIX bundle will follow the **unmodified** route.
```

> - Out: if at least one of your indicator of your stix bundle has triggered an observable creation, then the stix bundle will follow the **Out** route.

Reviewer: ```suggestion
- Out: if at least one of your indicators of your STIX bundle has triggered an observable creation, then the STIX bundle will follow the **Out** route.
```

#### Reduce Knowledge

Will filter out any entities in the current stage that do not match the filter conditions in the stage. 

> ##### Specifities of the component

Reviewer: ```suggestion
##### Specificities of the component

```

**Reduce will not work if the result of your reduce knowledge step is different from the entity triggering your playbook:**

If the result of the reduce knowledge ends up not matching the initial entity triggering yur playbook, then the reduce step will fail. As an example: 

> With a first step listening on: entity type = IPV4 OR Report AND label = test. And a step that reduce knowledge based on Entity type = IPV4. You will get the following results: 

Reviewer: ```suggestion
With a first step listening on: entity type = IPV4 OR Report AND label = test. And a step that reduces knowledge based on Entity type = IPV4. You will get the following results: 
```

- Test 1: edit a report that does not contain IPV4
   - Result: the playbook took the route "unmatched" (since no IP in the bundle, because triggering entity is the Report) 
- Test 2: edit a report that contains an IPV4
   - Result: the playbook took the route "unmatched" (since the entity triggering the playbook is the Report and not the IPV4)
- Test 3: edit an IPV4
   - Result: the playbook took the route "out" (since only the triggering entity is the IPV4)

 **Routes:**

> - Unmatched: if the bundle does not match the reduce condition, then the stix bundle will follow will follow the **unmatch** route. In this case, the playbook will act as the route "umatch" of the "match" component.

Reviewer: ```suggestion
- Unmatched: if the bundle does not match the reduce condition, then the stix bundle will follow the **unmatch** route. In this case, the playbook will act as the route "umatch" of the "match" component.
```

- Out: if your bundle is effectively reduced, then the stix bundle will follow the **Out** route.

#### Match knowledge

> Will allow you to continue only if some specific conditions are met. 

Reviewer: ```suggestion
It will allow you to continue only if some specific conditions are met. 
```

##### Specifities of the component
**Routes:**

> - No-match: if none of the entity/observable contained in the bundle passes the matching condition, then the stix bundle will follow will follow the **no-match** route.

Reviewer: ```suggestion
- No-match: if none of the entity/observable contained in the bundle passes the matching condition, then the STIX bundle will follow the **no-match** route.
```

> - Out: if at least one of the entity/observable contained in the bundle passes the matching condition, then the stix bundle will follow will follow the **out** route.

Reviewer: ```suggestion
- Out: if at least one of the entity/observable contained in the bundle passes the matching condition, then the STIX bundle will follow the **out** route.
```

At the top right of the interface, you can access execution trace of your playbook and consult the raw data after every step of your playbook execution.

> ### Useful for troublshooting:

Reviewer: ```suggestion
### Useful for troublshooting:

```

> - Less steps than the amount of steps of your playbook: If your playbook contains for instance 5 steps and only 4 steps are shown, it means that the playbook stopped at the 4th step.

Reviewer: ```suggestion
- Fewer steps than the number of steps of your playbook: If your playbook contains, for instance, 5 steps and only 4 steps are shown, it means that the playbook stopped at the 4th step.
```

> - Data created/ingested by the playbook does not contain the right modifications applied in the manipulate step: verify that your step is present, and that the operations you wanted to apply is well applied (you need to see the operation type, the field & the value)

Reviewer: ```suggestion
- Data created/ingested by the playbook does not contain the right modifications applied in the manipulate step: verify that your step is present, and that the operations you wanted to apply are well applied (you need to see the operation type, the field & the value)
```

> - Use the different routes to help troubleshoot your playbook: if you're using multiple components that involves mutliple routes, it is sometimes useful to test your playbook by adding a "manipulate knowledge" step to add a label for instance, in order to understand the route your bundle is taking if you feel that the data created by your playbook is not the one you expect. 

Reviewer: ```suggestion
- Use the different routes to help troubleshoot your playbook: if you're using multiple components that involves mutliple routes, it is sometimes useful to test your playbook by adding a "manipulate knowledge" step to add a label for instance, to understand the route your bundle is taking if you feel that the data created by your playbook is not the one you expect. 
```
